cache/policy_cache_test.go

```diff
@@ -0,0 +1,31 @@
+package cache
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/upfluence/pkg/cache/policy/size"
+)
+
+func TestPolicyCache(t *testing.T) {
+	c := WithEvictionPolicy(NewCache(), size.NewLRUPolicy(1))
+
+	done := make(chan struct{})
+
+	testHookEviction = func() { close(done) }
```

Comment:
I'm not sure to understand how it works

Reply:
as:
* the evictions asked by the policy are  executed in a background goroutine 
* the runtime execution is non-deterministic

i added a global variable functor that will be called one the eviction is made.
I use it here to notify the eviction by the policy as been done.

Reply:
Ok, you use it in the policy_cache, I forget that :+1: